app/src/main/java/com/transloadit/android_sdk/ActivityAssembly.java

```diff
@@ -1,74 +1,123 @@
 package com.transloadit.android_sdk;
 
-import android.app.Activity;
 import android.content.SharedPreferences;
 import android.net.Uri;
 
 import com.transloadit.sdk.Assembly;
+import com.transloadit.sdk.response.AssemblyResponse;
 
 import java.io.IOException;
 import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 import io.tus.android.client.TusAndroidUpload;
 import io.tus.android.client.TusPreferencesURLStore;
 import io.tus.java.client.ProtocolException;
 import io.tus.java.client.TusClient;
-import io.tus.java.client.TusExecutor;
 import io.tus.java.client.TusUpload;
-import io.tus.java.client.TusUploader;
 
 
 public class ActivityAssembly extends Assembly {
-    Activity activity;
-    Map<String, Uri> files;
+    AssemblyProgressListener listener;
+    Map<String, Uri> fileUris;
+    List<UploadTask> uploadTasks;
+    Transloadit transloadit;
+    AssemblyStatusUpdateTask statusUpdateTask;
+    String preferenceName;
+    String url;
```

Comment:
Is there a particular need for all of these attributes to be package private instead of private?